python_modules/dagster/dagster/core/storage/event_log/sql_event_log.py

```diff
@@ -261,60 +271,46 @@ def get_logs_for_run_by_log_id(
             )
 
         # adjust 0 based index cursor to SQL offset
-        query = query.offset(cursor + 1)
+        if isinstance(cursor, int):
+            query = query.offset(cursor + 1)
+        elif cursor:
+            try:
+                last_known_id = int(cursor)
+                query = query.where(SqlEventLogStorageTable.c.id > last_known_id)
+            except ValueError:
+                pass
```

Comment:
probably worth factoring out cursor -> filters to its own fn

confused by this since you are `str(offset)` basically in above callsites. I think a prefix scheme would help a lot for clarity on offset vs id vs timestamp variations.